components/brave_ads/core/internal/user_engagement/ad_events/search_result_ads/search_result_ad_event_handler_util.cc

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/internal/user_engagement/ad_events/search_result_ads/search_result_ad_event_handler_util.h"
+
+#include "base/check.h"
+#include "brave/components/brave_ads/core/internal/creatives/conversions/creative_set_conversion_builder.h"
+#include "brave/components/brave_ads/core/internal/creatives/conversions/creative_set_conversion_database_table_util.h"
+#include "brave/components/brave_ads/core/internal/creatives/conversions/creative_set_conversion_info.h"
+#include "brave/components/brave_ads/core/internal/settings/settings.h"
+#include "brave/components/brave_ads/core/mojom/brave_ads.mojom.h"  // IWYU pragma: keep
+#include "brave/components/brave_ads/core/public/ads_feature.h"
+
+namespace brave_ads {
+
+void MaybeBuildAndSaveCreativeSetConversion(
+    const mojom::CreativeSearchResultAdInfoPtr& mojom_creative_ad,
+    const mojom::SearchResultAdEventType event_type) {
+  const bool user_has_joined_rewards = UserHasJoinedBraveRewards();
+
+  const bool rewards_user_viewed_ad =
+      user_has_joined_rewards &&
+      event_type == mojom::SearchResultAdEventType::kViewedImpression;
+
+  const bool non_rewards_user_clicked_ad =
+      !user_has_joined_rewards &&
+      event_type == mojom::SearchResultAdEventType::kClicked;
+
+  const bool should_save =
+      rewards_user_viewed_ad || non_rewards_user_clicked_ad;
```

Comment:
should we process clicks for rewards users?

Reply:
Added comment. Thanks